project/src/js/ui/tournament/tournamentView.js

```diff
@@ -0,0 +1,152 @@
+    backButton(ctrl.tournament().fullName)
+  );
+  const bodyCtrl = tournamentBody.bind(undefined, ctrl);
+
+  return layout.free(headerCtrl, bodyCtrl, empty, empty);
+}
+
+function tournamentBody(ctrl) {
+  let data = ctrl.tournament();
+  if (!data) return null;
+
+  let body = null;
+  if (data.isFinished)
+    body = tournamentContentFinished(data);
+  else if (!data.isStarted)
+    body = tournamentContentCreated(data);
+  else
+    body = tournamentContentStarted(data);
+
+  return (m('.tournamentContainer', body));
+}
+
+function tournamentContentFinished(data) {
+  return (
+    <div>
+      { tournamentHeader(data, null, null)}
+      { tournamentLeaderboard(data, true) }
+    </div>
+  );
+}
+
+function tournamentContentCreated(data) {
+  return (
+    <div>
+      { tournamentHeader(data, data.secondsToStart, 'Starts in:')}
+      { tournamentLeaderboard(data, false) }
+    </div>
+  );
+}
+
+function tournamentContentStarted(data) {
+  return (
+    <div>
+      { tournamentHeader(data, data.secondsToFinish, '')}
+      { tournamentLeaderboard(data, false) }
+      { tournamentFeaturedGame(data) }
+    </div>
+  );
+}
+
+function tournamentHeader(data, time, timeText) {
+  let variant = variantInfo(data);
+  let control = timeControl(data);
+  return (
+    <div className='basicTournamentInfo'>
+      <strong> {variant + ' • ' + control + ' • ' + data.minutes + 'M' } </strong>
+      <div className='timeInfo'>
+        <strong> {timeInfo(time, timeText)} </strong>
+      </div>
+    </div>
+  );
+}
+
+function variantInfo(data) {
+  let variant = data.variant;
+  if(variant === 'standard') {
+    if(data.schedule)
+      variant = data.schedule.speed;
+    else if(data.position)
+      variant = data.position.name;
+    else
+      variant = '';
+  }
+
+  variant = variant.split(' ')[0]; // Cut off names to first word
+
+  if (variant.length > 0)
+    variant = variant.charAt(0).toUpperCase() + variant.substring(1);
+
+  return variant;
+}
+
+function timeControl(data) {
+  let limit = (data.clock.limit / 60);
+  if (data.clock.limit === 30)
+    limit = '½';
+  else if (data.clock.limit === 45)
+    limit = '¾';
+  return limit + '+' + data.clock.increment;
+}
+
+function timeInfo(time, preceedingText) {
+  if (!time) return '';
+
+  let timeStr = '';
+  let hours = Math.floor(time / 60 / 60);
+  let mins = Math.floor(time / 60) - (hours * 60);
+  let secs = time % 60;
+  if (hours > 0)
+    timeStr = preceedingText + ' ' + hours + ':' + pad(mins, 2) + ':' + pad(secs, 2);
+  else
+    timeStr = preceedingText + ' ' + mins + ':' + pad(secs, 2);
+  return timeStr;
+}
+
+function tournamentLeaderboard(data, showTrophies) {
+  return (
+    <div className='tournamentLeaderboard'>
+      <p className='tournamentTitle'> {i18n('leaderboard')} ({data.nbPlayers} Players)</p>
+      <table className='tournamentStandings'>
+        {data.standing.players.map(createLeaderboardItemRenderer(showTrophies))}
```

Comment:
Wrong use of `Array.map` here. 

It will be better explained here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

But shortly, you should pass only one arg to the `map` function, and this is the `map` callback. The second arg of map is and object that will be bound to the `this` of the callback function. I won't explain here all the subtleties of `this` in javascript, but I'm pretty sure you don't want to use that feature here.

A regular `Array.map` should be just like so:

```js
my.array.map(function(el, index) {
    // do stuff with el which is the element being mapped over
    // optionnaly do stuff with the index of element, but this is much less frequent usage
}); 

If you use `map` correctly you should never have to increment vars in your `map` callback. So I won't write a comment on all those lines like below

```js
podiumRank++;
```

because it's all related to the wrong `map` usage. I'm pretty sure that when you will see the right solutions for this problem it will appear much more obvious and elegant to you :)